{
  "id": "CVE-2024-4030",
  "sourceIdentifier": "cna@python.org",
  "published": "2024-05-07T21:15:09.467",
  "lastModified": "2024-05-07T21:15:09.467",
  "vulnStatus": "Received",
  "descriptions": [
    {
      "lang": "en",
      "value": "On Windows a directory returned by tempfile.mkdtemp() would not always have permissions set to restrict reading and writing to the temporary directory by other users, instead usually inheriting the correct permissions from the default location. Alternate configurations or users without a profile directory may not have the intended permissions.\n\nIf you\u2019re not using Windows or haven\u2019t changed the temporary directory location then you aren\u2019t affected by this vulnerability. On other platforms the returned directory is consistently readable and writable only by the current user.\n\nThis issue was caused by Python not supporting Unix permissions on Windows. The fix adds support for Unix \u201c700\u201d for the mkdir function on Windows which is used by mkdtemp() to ensure the newly created directory has the proper permissions.\n"
    }
  ],
  "metrics": {},
  "weaknesses": [
    {
      "source": "cna@python.org",
      "type": "Secondary",
      "description": [
        {
          "lang": "en",
          "value": "CWE-276"
        }
      ]
    }
  ],
  "references": [
    {
      "url": "https://github.com/python/cpython/issues/118486",
      "source": "cna@python.org"
    },
    {
      "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/PRGS5OR3N3PNPT4BMV2VAGN5GMUI5636/",
      "source": "cna@python.org"
    }
  ]
}